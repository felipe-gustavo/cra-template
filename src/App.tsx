import { useState } from "react";

import { AppContainer, AppHeader } from "./App.style";
import { Button } from "./components/atoms/Button";
import { AppDescription } from "./components/molecules/AppDescription";
import { AppLogo } from "./components/molecules/AppLogo";

function App() {
  const [spin, setSpin] = useState(false);

  const handlerToogleLogoSpin = () => {
    setSpin(!spin);
  };

  return (
    <AppContainer>
      <AppHeader>
        <AppLogo spin={!!spin} />
        <AppDescription />
        <Button onClick={handlerToogleLogoSpin} type="primary">
          {spin ? "Static logo" : "Spin logo"}
        </Button>
      </AppHeader>
    </AppContainer>
  );
}

export default App;
