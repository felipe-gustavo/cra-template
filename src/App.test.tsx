import { fireEvent, render, screen } from "@testing-library/react";

import App from "./App";

describe("<App />", () => {
  it("match with the screenshot", () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });

  it("starts the logo to static", () => {
    render(<App />);

    const button = screen.getByRole("button");

    expect(button.textContent).toBe("Spin logo");
  });

  it("is switching the logo to spin", () => {
    render(<App />);

    const button = screen.getByRole("button");
    fireEvent.click(button);

    expect(button.textContent).toBe("Static logo");
  });
});
