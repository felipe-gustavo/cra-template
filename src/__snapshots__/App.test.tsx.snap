// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> match with the screenshot 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ kIOTIl"
    >
      <header
        class="sc-gsnTZi xxDRN"
      >
        <div
          class="sc-eCYdqJ kSFQMJ"
        >
          <svg
            class="logo"
            height="40vmin"
            role="img"
          >
            logo.svg
          </svg>
        </div>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to reload.
        </p>
        <a
          class="sc-hKMtZM fHomBZ"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
        <button
          class="sc-dkzDqf fKwYlp primary"
          type="button"
        >
          Spin logo
        </button>
      </header>
    </div>
  </div>
</body>
`;
